docs/DeveloperGuide.md

```diff
@@ -1,38 +1,416 @@
+   of `XYZCommandParser`. This process is explained in further detail in the sequence diagrams below.
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+   ![Reference Frame Command Parser Sequence Diagram](https://raw.githubusercontent.com/AY2122s2-cs2113t-t10-1/tp/master/docs/images/developerguide/RefCommandParser.drawio.png)
 
-## Design & implementation
+   ![Reference Frame ParseABC Sequence Diagram](https://raw.githubusercontent.com/AY2122s2-cs2113t-t10-1/tp/master/docs/images/developerguide/RefParseABC.drawio.png)
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+   ![Reference Frame InvalidCommand Instantiation Sequence Diagram](https://raw.githubusercontent.com/AY2122s2-cs2113t-t10-1/tp/master/docs/images/developerguide/RefInvalidCommand.drawio.png)
 
+5. After `XYZCommandParser#getCommand` is called, `XYZCommandParser` prepares to create a `XYZCommand` object. 
+   To begin with, it parses all the _remaining arguments_ using `ParseABC` methods from the `Parser` class.
+   (`ParseABC` is a placeholder for specific methods in `Parser` class, 
+   e.g. `Parser#parseName` and `Parser#parseSessionId`)
+   * For example, `SessionCreateCommandParser` has to call `parsePersonList`, `parseGroupId`, `parseName` and
+      `parseLocalDate` from `Parser` class in order to get the details to create a `Session` object.
+   * If an exception is encountered, `XYZCommandParser` handles the exception accordingly, and if necessary,
+      throws an exception back to `Parser` class, resulting in an `InvalidCommand` object being created and returned.
+6. In detail, when `Parser#parseABC` is called, `Parser` class calls the method `getArgumentFromDelimiter` from
+   `ParserUtils` class, which returns the respective object being parsed.
+   * For example, when `SessionCreateCommandParser` calls `Parser#parsePersonList`,
+      `ParserUtils#getArgumentFromDelimiter` is called. After returning a `String` object containing the arguments to
+      `Parser` class, `Parser` class returns a `String[]` object to `SessionCreateCommandParser` after processing the
+      arguments.
+   * Any exception encountered by `ParserUtils` class is propagated back to `XYZCommandParser` to be handled.
+7. After all necessary information is parsed, `XYZCommandParser` instantiates a new `XYZCommand` object and passes
+   all parsed information to it through the constructor.
+8. The newly instantiated `XYZCommand` object is then returned from `XYZCommandParser` to `Parser` class,
+   and finally back to `SplitLah` to be run.
+
+### Add a session
+**API reference:** [`SessionCreateCommand.java`](https://github.com/AY2122S2-CS2113T-T10-1/tp/blob/master/src/main/java/seedu/splitlah/command/SessionCreateCommand.java)
```

Comment:
I like that you've coloured classes and objects according to their role:
![image](https://user-images.githubusercontent.com/69501969/160958539-7ecb11b2-39c6-4315-a6a2-4bf79d47fd6b.png)
![image](https://user-images.githubusercontent.com/69501969/160958556-49b5b71c-f5d2-4c7e-993e-2ee080e6bcd8.png)
However, some diagrams such as this one does not follow the same formatting:
![image](https://user-images.githubusercontent.com/69501969/160958615-4ad67bc4-491a-4796-8b1f-d10ff14c26cd.png)
Perhaps the document would be more cohesive with consistent formatting.